### PR TITLE
Fix panic when reloading a shrunk file

### LINKED ISSUE #2305: Panic when reloading a file that has shrunk
### Summary

If you reload a file while your cursor is out-of-bounds of the reloaded file, a panic occurs.

Backtrace:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Line index out of bounds: line index 4569, Rope/RopeSlice line count 2923', /build/helix-22.03-vendor.tar.gz/ropey/src/slice.rs:424:41
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: ropey::slice::RopeSlice::line_to_byte
   4: helix_term::ui::editor::EditorView::doc_syntax_highlights
   5: helix_term::ui::editor::EditorView::render_view
   6: <helix_term::ui::editor::EditorView as helix_term::compositor::Component>::render
   7: helix_term::compositor::Compositor::render
   8: helix_term::application::Application::handle_terminal_events
   9: helix_term::application::Application::event_loop::{{closure}}
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  11: tokio::park::thread::CachedParkThread::block_on
  12: tokio::runtime::thread_pool::ThreadPool::block_on
  13: tokio::runtime::Runtime::block_on
  14: hx::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### Reproduction Steps

1. Open a file and go to line N (N must be at least some number. It works for me at ~200).
2. Externally edit the file so that it is much shorter than N
3. `:reload` the file.

### Helix log

There is no relevant log information.

### Platform

Linux

### Terminal Emulator

gnome-terminal

### Helix Version

helix 22.03 (d4e45fd4)

### PR DESCRIPTION
**Pull request by pppKin**
a small PR to fix #2305

### CODE REVIEW

File: helix-term/src/commands/typed.rs [hunk 1]
```diff
@@ -782,7 +782,9 @@ fn reload(
     _event: PromptEvent,
 ) -> anyhow::Result<()> {
     let (view, doc) = current!(cx.editor);
-    doc.reload(view.id)
+    doc.reload(view.id).map(|_| {
+        view.ensure_cursor_in_view(doc, 0);
+    })
```
Comment (archseer) [inline helix-term/src/commands/typed.rs:787]:
```suggestion
    let config = cx.editor.config();
    doc.reload(view.id).map(|_| {
        view.ensure_cursor_in_view(doc, config.scrolloff);
    })
```

Comment (pppKin) [reply helix-term/src/commands/typed.rs]: Good idea; pushed

**Comment by pppKin** [comment]
As for the lints, I ran `cargo clean` and `cargo clippy -- -D warnings` on my machine but no warnings/errors whatsoever.  So I'm confused as well.

**Comment by the-mikedavis** [comment]
It looks like rust (and clippy) 1.61.0 were released recently and the CI has already started using it. https://github.com/helix-editor/helix/commit/e04bb8b8915bfe1df1a5ee7a8750f2589f2aae06 was just merged that fixes the lints so if you rebase on top of that, the lints CI should clear up

**Comment by pppKin** [comment]
Ah, I updated my rust toolchain to 1.61 and indeed it would generate erros now. I have rebased onto latest master.

**Pull request merged**